### PR TITLE
[Common] util to compare dimensions

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Add util function to compare tensor dimensions.

### CODE REVIEW

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 1]
```diff
@@ -1087,6 +1083,27 @@ gst_tensor_dimension_is_valid (const tensor_dim dim)
...
+  guint i;
+
+  /* Do not compare zero dimension. */
+  if (dim1[0] == 0U && dim2[0] == 0U)
+    return FALSE;
+
+  for (i = 0; i < NNS_TENSOR_RANK_LIMIT; i++) {
+    if (dim1[i] != dim2[i])
+      return FALSE;
+  }
```
Comment (myungjoo) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:1102]: with 0-init, we may have 4:3:1:00000 and 4:3:000000. Are they different? or same?

Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:1102]: 1. After implementing 0-init dims, 4:3:1:00 and 4:3:00 will be same.
2. Current dimension is filled with 1, and this util function should be updated later.
e.g., compare remained dim like below, (I didnt add below lines in this PR.)
```
    if (dim1[i] != dim2[i]) {
      tensor_dim remained;

      if (dim1[i] == 0)
        remained = dim2;
      else if (dim2[i] == 0)
        remained = dim1;
      else
        return FALSE;

      /* Supposed dimension is same if remained dimension is 1. */
      for (; i < NNS_TENSOR_RANK_LIMIT; i++) {
        if (remained[i] > 1)
          return FALSE;
      }
    }
```

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 2]
```diff
@@ -1009,6 +1005,27 @@ gst_tensor_dimension_is_valid (const tensor_dim dim)
...
+ * @brief Compare the tensor dimension.
+ * @return TRUE if given tensors have same dimension.
+ */
+gboolean
+gst_tensor_dimension_is_equal (const tensor_dim dim1, const tensor_dim dim2)
+{
+  guint i;
+
+  /* Do not compare zero dimension. */
+  if (dim1[0] == 0U && dim2[0] == 0U)
```
Comment (songgot) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:1018]: Instead of the operator '&&', maybe 'II'?

**Review by myungjoo — APPROVED** [review]

**Review by anyj0527 — APPROVED** [review]
LGTM 👍

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_plugin_api_util_impl.c]: updated

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 3]
```diff
@@ -209,16 +207,14 @@ gst_tensor_info_is_equal (const GstTensorInfo * i1, const GstTensorInfo * i2)
...
-      return FALSE;
-    }
+  if (!gst_tensor_dimension_is_equal (i1->dimension, i2->dimension)) {
+    gchar *dim_str1 = gst_tensor_get_dimension_string (i1->dimension);
+    gchar *dim_str2 = gst_tensor_get_dimension_string (i2->dimension);
+    nns_logd ("Tensor info is not equal. Given tensor dimensions %s vs %s",
+        dim_str1, dim_str2);
+    g_free (dim_str1);
+    g_free (dim_str2);
+    return FALSE;
```
Comment (wooksong) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:217]:
```suggestion
  if (!gst_tensor_dimension_is_equal (i1->dimension, i2->dimension)) {
    g_autofree gchar *dim_str1 = gst_tensor_get_dimension_string (i1->dimension);
    g_autofree gchar *dim_str2 = gst_tensor_get_dimension_string (i2->dimension);

    nns_logd ("Tensor info is not equal. Given tensor dimensions %s vs %s",
        dim_str1, dim_str2);
    return FALSE;
```

Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:217]: I will update this in other PR.

File: gst/nnstreamer/nnstreamer_plugin_api_util_impl.c [hunk 4]
```diff
@@ -1131,10 +1152,8 @@ gst_tensor_dimension_string_is_equal (const gchar * dimstr1,
     if (!rank1 || !rank2)
       goto done;
 
-    for (j = 0; j < NNS_TENSOR_RANK_LIMIT; j++) {
-      if (dim1[j] != dim2[j])
-        goto done;
-    }
+    if (!gst_tensor_dimension_is_equal (dim1, dim2))
```
Comment (wooksong) [inline gst/nnstreamer/nnstreamer_plugin_api_util_impl.c:1155]: ```rank1``` and ```rank2``` are gathered from ```dim1``` and ```dim2```. How about checking the equivalent of them before getting ```rank1``` and ```rank2```?

**Pull request merged**